### PR TITLE
v0.12.0 Updates and Changelog

### LINKED ISSUE #465: Collection Metadata Database Column
From #458 we need to add a new database column of `metadata = '{}'::json` and remove the pydantic `Optional` to ease migration and prevent conflicts in the future.

### PR DESCRIPTION
**Pull request by dgasmith**
## Description
Updates the changelog and fixes #465. Items beyond #465:

- Alters server logging to occur in the background and uses much faster but more imprecise algorithms so very large datasets do not hurt database performance.
- Adds a Molecule benchmark script.
- Validates v0.11.0 -> v0.12.0 migration scripts.

## Status
- [x] Changelog updated
- [x] Ready to go

### CODE REVIEW

File: docs/qcfractal/source/changelog.rst [hunk 1]
```diff
@@ -15,25 +15,47 @@ Changelog
 .. Bug Fixes
 .. +++++++++
 
-0.12.0 / 2019-MM-DD
+0.12.0 / 2019-11-06
 -------------------
 
+Highlights:
```
Comment (mattwelborn) [inline docs/qcfractal/source/changelog.rst]:
```
Highlights
+++++++
```
?

File: docs/qcfractal/source/changelog.rst [hunk 2]
```diff
@@ -15,25 +15,47 @@ Changelog
...
+
+- The ability to handle very large datasets (1M+ entries) quickly and efficiently.
+- Store and compute Wavefunction information.
+- Build and serve views for Datasets.
```
Comment (mattwelborn) [inline docs/qcfractal/source/changelog.rst]: Build, serve, and export

File: docs/qcfractal/source/changelog.rst [hunk 3]
```diff
@@ -15,25 +15,47 @@ Changelog
...
+
 New Features
 ++++++++++++
 - (:pr:`433`) ``Dataset`` and ``ReactionDataset`` (``interface.collections``) now have a ``download``` method which
   downloads a frozen view of the dataset. This view is used to speed up calls to ``get_values``, ``get_molecules``,
   ``get_entries``, and ``list_values``.
+- (:pr:`440`) Wavefunctions can now be stored in the database using Result ``protocols``.
+- (:pr:`453`) The server now periodically logs manager and current state to provide data over time.
+- (:pr:`460`) Contributed values are now in their own table to speed up access of Collections.
+- (:pr:`461`) Services now update their correspond record every iteration.
```
Comment (mattwelborn) [inline docs/qcfractal/source/changelog.rst]: corresponding. Also, give an example of a service.

File: docs/qcfractal/source/changelog.rst [hunk 4]
```diff
@@ -15,25 +15,47 @@ Changelog
...
+Highlights:
+
+- The ability to handle very large datasets (1M+ entries) quickly and efficiently.
+- Store and compute Wavefunction information.
+- Build and serve views for Datasets.
+- A new GUI dashboard to observe the current state of the server, see statistics, and fix issues.
+
 New Features
 ++++++++++++
 - (:pr:`433`) ``Dataset`` and ``ReactionDataset`` (``interface.collections``) now have a ``download``` method which
```
Comment (mattwelborn) [inline docs/qcfractal/source/changelog.rst]: Add pr 462 to this list, remove from Enhancements, don't mention the progress bar.

File: docs/qcportal/source/changelog.rst [hunk 5]
```diff
@@ -15,6 +15,40 @@ Changelog
...
+Enhancements
+++++++++++++
+- (:pr:`429`) Enables protocols for ``OptimizationDataset`` collections.
+- (:pr:`430`) Adds additional QCPortal type hints.
+- (:pr:`433`, :pr:`443`) ``Dataset`` and ``ReactionDataset`` (``interface.collections``) are now faster for calls to calls to ``get_values``, ``get_molecules``,
+  ``get_entries``, and ``list_values`` for large datasets if the server is configured to use frozen views. See "Server-side Dataset Views" documentation. Subsets
+  may be passed to ``get_values``, ``get_molecules``, and ``get_entries``
+- (:pr:`447`) Enables the creation of plaintext (xyz and csv) output from Dataset Collections.
+- (:pr:`458`) Collections now have a metadata field.
+- (:pr:`462`) Dataset downloads now have a TQDM progress bar.
```
Comment (mattwelborn) [inline docs/qcportal/source/changelog.rst]: Same as above

File: qcfractal/alembic/versions/a2f76bb7be65_updates_collections_table_with_owner_.py [hunk 6]
```diff
@@ -18,8 +18,14 @@
 def upgrade():
     # ### commands auto generated by Alembic - please adjust! ###
     op.add_column('collection', sa.Column('description', sa.String(), nullable=True))
-    op.add_column('collection', sa.Column('group', sa.String(length=100), nullable=True))
-    op.excecute("UPDATE collection SET group=default")
+
+    op.add_column('collection', sa.Column('group', sa.String(), nullable=False, server_default='default'))
```
Comment (mattwelborn) [inline qcfractal/alembic/versions/a2f76bb7be65_updates_collections_table_with_owner_.py]: what does `op.alter_column('collection', 'group', server_default=None)` do?

File: qcfractal/server.py [hunk 7]
```diff
@@ -544,17 +548,16 @@ def update_public_information(self) -> None:
...
-        update = {
-            "counts": {
-                "collection": data["collection_count"],
-                "molecule": data["molecule_count"],
-                "result": data["result_count"],
-                "kvstore": data["kvstore_count"],
-            }
-        }
+        data = self.storage.get_server_stats_log(limit=1)["data"]
+
```
Comment (mattwelborn) [inline qcfractal/server.py]: do you need a default value for kvstore here?

File: qcfractal/storage_sockets/sqlalchemy_socket.py [hunk 8]
```diff
@@ -282,22 +282,23 @@ def get_limit(self, limit: Optional[int]) -> int:
 
         return limit if limit and limit < self._max_limit else self._max_limit
 
-    def get_query_projection(self, className, query, projection, limit, skip, exclude=None):
+    def get_query_projection(self, className, query, *, limit=None, skip=0, include=None, exclude=None):
         # Todo: projection to be renamed to include_only
```
Comment (mattwelborn) [inline qcfractal/storage_sockets/sqlalchemy_socket.py]: remove TODO comment

File: qcfractal/server.py (hunk 7 above)
Comment (dgasmith) [reply qcfractal/server.py]: Yup.

File: qcfractal/alembic/versions/a2f76bb7be65_updates_collections_table_with_owner_.py (hunk 6 above)
Comment (dgasmith) [reply qcfractal/alembic/versions/a2f76bb7be65_updates_collections_table_with_owner_.py]: For some reasons strings do not work here, it thinks we are trying to set a composite column (similar to a C struct). So we set the column with a server default meaning that any `None` will be replaced by that value (default on creation) and then we turn off the default forcing pydantic to provide it. 

Similar operation to the `UPDATE ... SET ...` calls.

**Pull request merged**